src/api/data/cache/cache-helper.go

```diff
@@ -0,0 +1,212 @@
+package cache
```

Comment:
This filename isn't really idiomatic, I suggest dropping the cache prefix and going with `helpers.go` or `tar_helpers.go` or `cache_helper.go`.